### ISSUE TITLE
`Poly.unit` is not working

### ISSUE THREAD

**Issue by sylee957** [sympy/sympy#21314]
```
u = Poly([1, 2, 3], x)
u.unit
```
Gives `AttributeError: 'DMP' object has no attribute 'unit'`

**Comment by oscarbenjamin** [comment]
The docstring says:
```
        """Return unit polynomial with ``self``'s properties. """
```
I don't understand what that even means.

**Comment by guptapratykshh** [comment]
The domain argument is unnecessary there because the domain is already embedded in the DMP `(self.rep.dom)`. The new method does not accept a domain keyword argument, so it can be removed. It can be this  -

`def unit(self):`
`return self.new(DMP.one(self.rep.lev, self.rep.dom), *self.gens)`

**Comment by guptapratykshh** [comment]
Thank you for the suggestion. I agree that having both `Poly.unit` and `Poly.one` is redundant, and it makes sense that to remove `Poly.unit` in favour of `Poly.one`. Based on this i will make a PR on this.

**Comment by sylee957** [comment]
I agree. I don't think that we need to worry about backward compatibility, if poly.unit is already bugged

**Comment by guptapratykshh** [comment]
Thank you for the clarification. Since `Poly.unit` is already bugged and not functional, I agree that backward compatibility is not a concern. I’ve proceeded with removing the unit method entirely.
